File: java2csharp/java/Microsoft.Mobile.Engagement/engagement/reach/EngagementAbstractAnnouncement.java
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

package com.microsoft.azure.engagement.reach;

import static com.microsoft.azure.engagement.reach.ContentStorage.ACTION_URL;

import java.util.Map;

import org.json.JSONException;

import android.content.ContentValues;

/** Base class for all kind of announcements. */
public abstract class EngagementAbstractAnnouncement extends EngagementReachInteractiveContent
{
  /** Action's URL */
  private final String mActionURL;

  /**
   * Parse an announcement.
   * @param campaignId already parsed campaign id.
   * @param values campaign data.
   * @param params special parameters to inject in the action URL of the announcement.
   * @throws JSONException if a parsing error occurs.
   */
  EngagementAbstractAnnouncement(CampaignId campaignId, ContentValues values,
    Map<String, String> params) throws JSONException
  {
    super(campaignId, values);
    String actionURL = values.getAsString(ACTION_URL);
    for (Map.Entry<String, String> param : params.entrySet())
    {
      if (actionURL != null)
        actionURL = actionURL.replace(param.getKey(), param.getValue());
    }
    mActionURL = actionURL;
  }

  /**
   * Get action's URL.
   * @return action's URL.
   */
  public String getActionURL()
  {
    return mActionURL;
  }
}

File: java2csharp/csharp/Microsoft.Mobile.Engagement/engagement/reach/EngagementAbstractAnnouncement.cs
using System.Collections.Generic;

/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

namespace com.microsoft.azure.engagement.reach
{

//JAVA TO C# CONVERTER TODO TASK: This Java 'import static' statement cannot be converted to C#:
//	import static com.microsoft.azure.engagement.reach.ContentStorage.ACTION_URL;

	using JSONException = org.json.JSONException;

	using ContentValues = android.content.ContentValues;

	/// <summary>
	/// Base class for all kind of announcements. </summary>
	public abstract class EngagementAbstractAnnouncement : EngagementReachInteractiveContent
	{
	  /// <summary>
	  /// Action's URL </summary>
	  private readonly string mActionURL;

	  /// <summary>
	  /// Parse an announcement. </summary>
	  /// <param name="campaignId"> already parsed campaign id. </param>
	  /// <param name="values"> campaign data. </param>
	  /// <param name="params"> special parameters to inject in the action URL of the announcement. </param>
	  /// <exception cref="JSONException"> if a parsing error occurs. </exception>
//JAVA TO C# CONVERTER WARNING: Method 'throws' clauses are not available in .NET:
//ORIGINAL LINE: EngagementAbstractAnnouncement(CampaignId campaignId, android.content.ContentValues values, java.util.Map<String, String> params) throws org.json.JSONException
	  internal EngagementAbstractAnnouncement(CampaignId campaignId, ContentValues values, IDictionary<string, string> @params) : base(campaignId, values)
	  {
		string actionURL = values.getAsString(ACTION_URL);
		foreach (KeyValuePair<string, string> param in @params.SetOfKeyValuePairs())
		{
		  if (actionURL != null)
		  {
			actionURL = actionURL.Replace(param.Key, param.Value);
		  }
		}
		mActionURL = actionURL;
	  }

	  /// <summary>
	  /// Get action's URL. </summary>
	  /// <returns> action's URL. </returns>
	  public virtual string ActionURL
	  {
		  get
		  {
			return mActionURL;
		  }
	  }
	}

}